OREData/ored/portfolio/asianoption.cpp

```diff
@@ -108,17 +107,11 @@
      QL_REQUIRE(processType != "", "ProcessType must be configured, this is unexpected");
 
      if (!isEuropDeferredCS) {
-         // If the option is not a European deferred CS contract, i.e. the standard European/American physical/cash-settled ones. 
+         // If the option is not a European deferred CS contract, i.e. the standard European physical/cash-settled ones.
          if (processType == "Discrete") {
-             Real runningAccumulator = 0;
-             if (averageType == Average::Type::Geometric) {
-                 runningAccumulator = 1;
-             }
-             Size pastFixings = 0;
-
              Schedule observationSchedule;
-             if (observationDates_.hasData())
-                 observationSchedule = makeSchedule(observationDates_);
```

Comment:
Surely this was `if` for a reason?

Reply:
I actually implemented that myself, looking at https://github.com/OpenSourceRisk/Engine/pull/75/files, but I'm quite certain it wouldn't even run in its current form. The `Schedule` isn't default constructed so there'd be no dates to query when it doesn't get initialized, but there's no test for that special case. The proposed change will force the data to be populated so that we always have some dates to use.